src/infinitemps.jl

```diff
@@ -65,9 +67,44 @@ getsite(i::Index) = getsite(tags(i))
 function ITensors.findfirstsiteind(ψ::InfiniteMPS, i::Index)
   c = ITensorInfiniteMPS.getcell(i)
   n1 = getsite(i)
-  return (c - 1) * nsites(ψ) + n1
+  if translater(ψ) == translatecell
+    return (c - 1) * nsites(ψ) + n1
+  else
```

Comment:
Could you explain this section of code?

I guess you are looping through the indices and seeing which site matches, up to some limit of 1000 sites? I remember thinking I would have to design it that way until I thought of just using the tag but I could see that is no longer possible with a general translation function.

Is there some assumption about the relationship between the tags and the `translater` function? Since it still looks like you are making use of `getcell` and `getsite` in the general case.

Reply:
Yes, the idea is that this way one can deal with more general tags and it allows for an easy benchmarking: namely replacing translation c->c+1 by c->c+2.

The original version assumed that for the tags, translation by one unit cell is c->c+1
Here we just assume that c is an increasing function.

I am not sure it is actually a useful feature to be honest. It might be worth putting the extension in a test, and just specifying that the tag pattern for translation must remain the same. (In fact thinking about it, it is probably the best solution, I will fix it like that if you agree).

I updated the getsite function (or at least the getsite) to be a bit more generic in order to treat this kind of edge states. It can be rolled back easily.

Reply:
Yeah, it sounds reasonable to assume the `translator` functions shifts the cell tags by 1 for now, and if we come across a more general case than we can generalize it later.

Reply:
Sure. I will modify the code.